third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -506,12 +512,18 @@ findPipelineableLoads(scf::ForOp forOp,
               }
               auto pointeeTy = cast<tt::PointerType>(tensorTy.getElementType())
                                    .getPointeeType();
-              // If the max continugous bits we can read is < 32, buffer in
-              // registers.
-              if (vecContiguity * pointeeTy.getIntOrFloatBitWidth() >= 32) {
+              unsigned width =
+                  vecContiguity * pointeeTy.getIntOrFloatBitWidth();
+              // Limit shared memory sharing to width >= 32 elements.
+              LDBG("Load " << *loadOp << " has width " << width);
+              if (width >= 32) {
                 sharedEncoding =
                     getSharedEncIfAllUsersAreDotEnc(op->getResult(0))
                         .value_or(nullptr);
+              } else if (width < 32 &&
```

Comment:
I think now this becomes `else` you don't need `width < 32 &&` part anymore. :)

Reply:
ofc i saw this but as a personal rule i don't like conditionals that have non-intersecting conditions because they depend on execution order. so if you remove the `width < 32` check then you only have a check for isafamily and if someone refactors the `if else` you will get a bug where (possibly) the second else does or doesn't trigger independently of `width`.

Reply:
It's common to have `if (..) else if (..) else if (..) ..` structures in various codebases to build sequence of checks with the following cases assuming the previous ones not hit--that's the purpose of `else` right? Putting extra `width < 32` there is just dead code effectively--the C++ compiler can probably optimize it away but still redundant information to me in the source code.

Reply:
done